### PR TITLE
Fix ntfy

### PR DESCRIPTION
**Pull request by joachifm**
###### Things done:
- [x] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- [x] Built on platform(s): Linux
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### More

The results of my attempt at fixing `pythonPackages.ntfy`. I'm not a python expert, so beware.
Built both `python2Packages.ntfy` & `python3Packages.ntfy`.

cc ntfy maintainer @kamilchm

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -12900,6 +12898,26 @@ in modules // {
...
+    };
+
+    buildInputs = with self; [ nose ];
+
+    meta = {
+      description = "Emoji for Python";
+      homepage = https://pypi.python.org/pypi/emoji/;
+      license = licenses.bsd3;
+      maintainers = with maintainers; [ joachifm ];
+      platforms = platforms.all;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: `buildPythonPackage` uses `meta.platforms` from the interpreter, which is `platforms = with stdenv.lib.platforms; linux ++ darwin;`, so there is no need to set `platforms`.

Comment (joachifm) [inline pkgs/top-level/python-packages.nix]: Fixed

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -12901,6 +12899,25 @@ in modules // {
...
+  emoji = buildPythonPackage rec {
+    name = "emoji-${version}";
+    version = "0.3.9";
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/e/emoji/${name}.tar.gz";
+      sha256 = "19p5c2nlq0w9972rf9fghyswnijwgig5f8cyzs32kppnmzvzbkxw";
+    };
+
+    buildInputs = with self; [ nose ];
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: did you check whether there are actually tests and these were run? Quite often checkPhase succeeds with zero tests ran.

Comment (joachifm) [inline pkgs/top-level/python-packages.nix]: Unless checkphase runs nose, I suppose the tests didn't execute.

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: checkPhase doesn't explicitly run nose but instead runs `python setup.py test`. It is possible that nose is specified as test suite in setup.py in which case it is *supposed* to run. But well, with Python packages you're never quite sure what is going to happen. Therefore, it's best to check whether you see that a certain amount of tests have run or not.

**Comment by kamilchm** [comment]
What errors does it fix? I can help with it but I need to know how to test it. I use it for few days without errors, but I don't use any extension, so I suspect that they cause some issues?

**Comment by joachifm** [comment]
@kamilchm this PR fixes the build itself. See https://hydra.nixos.org/build/33157748

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (joachifm) [inline pkgs/top-level/python-packages.nix]: I've specified checkPhase explicitly, now it runs the tests.

**Comment by kamilchm** [comment]
How can I reproduce it on my machine?
Is there a way to get failure notification from hydra for owned PRs?

**Comment by joachifm** [comment]
@kamilchm set `build-use-sandbox = true` in `nix.conf` or otherwise ensure that the build is sandboxed (I think that's why you're not seeing errors on your end). Hydra used to send build notifications, but does not appear to do so anymore (it got a bit spammy, that might be why).

**Comment by kamilchm** [comment]
Is this the same as {{nix-build --option build-use-sandbox true}}? I tried it and I don't get any errors.

I see that on my machine some dependencies are downloaded from *pypi* while hydra rejects it. It would be a lot easier when errors like these  would be repeatable. Right now I see passing build on my machine, [on travis](https://travis-ci.org/NixOS/nixpkgs/builds/114714151) and [hydra](https://hydra.nixos.org/build/33064171) :/

**Comment by joachifm** [comment]
Note that the hydra build I linked is newer than what you're referring to. I also reproduced the error on my machine. Not sure what's going on. It's pretty clear that ntfy does try to download stuff during the build, however, which will fail in a sandboxed environment.

**Comment by globin** [comment]
> Is this the same as `nix-build --option build-use-sandbox true`? I tried it and I don't get any errors.

Only if you don't have the nix daemon running. So it won't work on NixOS where you have to use nix.useChroot

**Comment by kamilchm** [comment]
@joachifm I agree that this build should fail. I just try to find a way to make sure that I don't introduce similar issues in the future. I passed 3 false positives on the way and I get info about failure few days after merge. If I knew about it before we could save your time. But it's more general CI issue than this PR.

Thx for your work on this :+1: 

@globin finally, I get failing build after I change it in NixOS config. Why is it disabled by default?

**Comment by globin** [comment]
@kamilchm https://github.com/NixOS/nix/issues/179

**Pull request merged**